spark/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -608,6 +608,10 @@ object Checkpoints
       deltaLog: DeltaLog,
       snapshot: Snapshot): LastCheckpointInfo = recordFrameProfile(
       "Delta", "Checkpoints.writeCheckpoint") {
+    if (spark.conf.get(DeltaSQLConf.DELTA_WRITE_CHECKSUM_ENABLED)) {
+      snapshot.validateChecksum(Map("context" -> "writeCheckpoint"))
+    }
```

Comment:
Should this check be done in the beginning or in the end?

Reply:
I feel that it would be a waste of time to write a checkpoint only to find out that there is something wrong with the delta log.

Reply:
@prakharjain09 Since most of this PR is about the validation logic itself, maybe we can address the question of when validation should be triggered in a follow up?